### PR TITLE
Support for connect native services in topology view.

### PR DESCRIPTION
**Pull request by apollo13**
This is a draft implementation for #12096. Misses tests and probably cleanups of the JS code :)

### CODE REVIEW

**Comment by Amier3** [comment]
Hey @apollo13 

Thanks for raising this PR! Just letting you know that we didn't skip over this, our front-end eng will be looking over this and giving some guidance on the tests either this week or the next 👍

**Comment by johncowen** [comment]
Hey @apollo13 👋 

Quick note to let you know I'll be picking this up, I'll need a little time to look over the linked issues and get a handle on things so I'll probably be back in a couple o' days or so with questions/queries, just wanted to let you know it's on my radar

ty!

**Comment by johncowen** [comment]
Hey @apollo13 👋 

Apologies for the delay in coming back to you here. As most of the changes here are golang related to I've asked somebody on the team to swing by and give this a 👁️ .

From the JS side, as there are very minor changes I'd be happy for this to be merged as is (if the go bits are fine) and we can take it from there, so I'll leave approval up to the go folks.

File: agent/consul/state/catalog.go [hunk 1]
```diff
@@ -3573,8 +3578,8 @@ func (s *Store) ServiceTopology(
 		if downstream.Service.Kind == structs.ServiceKindConnectProxy {
 			sn = structs.NewServiceName(downstream.Service.Proxy.DestinationServiceName, &downstream.Service.EnterpriseMeta)
 		}
-		if _, ok := tproxyMap[sn]; !ok && downstreamSources[sn.String()] != structs.TopologySourceRegistration {
-			// If downstream is not a transparent proxy, remove references
+		if _, ok := tproxyMap[sn]; (!ok && !downstream.Service.Connect.Native) && downstreamSources[sn.String()] != structs.TopologySourceRegistration {
```
Comment (kisunji) [inline agent/consul/state/catalog.go:3581]: Is there a reason for `()` around ```(!ok && !downstream.Service.Connect.Native)```?

**Review by kisunji — COMMENTED** [review]
LGTM!

Would you be able to update a test at the handler level here:
https://github.com/hashicorp/consul/blob/main/agent/ui_endpoint_test.go#L998
I think all it needs is an assertion for a mock `NodeService` with `svc.Connect.Native: true`

File: agent/consul/state/catalog.go (hunk 1 above)
Comment (apollo13) [reply agent/consul/state/catalog.go:3581]: No, probably a left-over -- will fix.

**Comment by apollo13** [comment]
@kisunji I think I got a test running; I added a new node for it because otherwise those would be a mess…

**Comment by apollo13** [comment]
What I do not know yet: Should I include the static assets (`agent/uiserver/bindata_assetfs.go`) or not?

**Comment by kisunji** [comment]
> What I do not know yet: Should I include the static assets (`agent/uiserver/bindata_assetfs.go`) or not?

Ah you can skip that; we have CI to automatically generate them

**Review by kisunji — APPROVED** [review]
Looks like the test ended up being more of a hassle than I expected, but thank you for being so thorough!

**Comment by kisunji** [comment]
Waiting on the team to give a final 👍 before merging

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/586713.

**Pull request merged**